core/src/main/scala/org/apache/spark/rdd/PairRDDFunctions.scala

```diff
@@ -103,13 +105,42 @@ class PairRDDFunctions[K, V](self: RDD[(K, V)])
   }
 
   /**
-   * Simplified version of combineByKey that hash-partitions the output RDD.
+   * Use [[combineByKeyWithClassTag]] instead.
+   */
+  def combineByKey[C](createCombiner: V => C,
+                      mergeValue: (C, V) => C,
+                      mergeCombiners: (C, C) => C,
+                      partitioner: Partitioner,
+                      mapSideCombine: Boolean = true,
+                      serializer: Serializer = null): RDD[(K, C)] = {
+    log.warn("Running combineByKey. Shuffle will not have combine value ClassTag. " +
```

Comment:
Agree. I'll update this to remove the warning message and revert the documentation.

Shuffle managers that don't depend on having a defined combiner class can throw an error, log a warning, etc.